### PR TITLE
Merged unittests, python-package and e2e into single workflow - now e2e is run only if unittests pass

### LINKED ISSUE #164: Chain e2e tests to only run after unit tests succeed

### PR DESCRIPTION
**Pull request by Lasica**
#### Description
Now e2e execution can be conditional on unittest success. Also added filters for running the test (only python files outside docs trigger the execution)

Resolves #164

##### PR Checklist
- [x] [Changelog](CHANGELOG.md) updated

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -1,15 +1,60 @@
...
 on: # workflow_dispatch
   push:
-     branches:
-       - develop
+    branches:
+      - develop
+    paths: 
+      - "**.py"
+      - "!docs/**"
+      - ".github/workflows/test.yml"
```
Comment (marrrcin) [inline .github/workflows/test.yml:10]: Maybe also add `tests/e2e/` path to trigger here?

Comment (em-pe) [reply .github/workflows/test.yml:10]: I have a mixed feelings restricting triggers to run only on modified paths, you can never be sure if those were not run due to the condition or if there's misconfiguration of some sort. If you're changing only documentation you can always add `[no ci]` to the commit message that will skip the actions execution, but you do this consciously. 

Currently the costs are not such big of a deal to worry about that, and there's not much money saved: 
* we don't pay for github runners
* currently kfp cluster has a fixed size, so we pay even if it doesn't run tests

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -1,15 +1,60 @@
-name: End 2 End Tests (Spaceflights Quickstart)
+name: Unit and E2E Tests workflow (Spaceflights Quickstart)
```
Comment (em-pe) [inline .github/workflows/test.yml:1]: Just `Tests` is sufficient here I guess

File: .github/workflows/test.yml (hunk 1 above)
Comment (Lasica) [reply .github/workflows/test.yml:10]: how about only exclusion then?
path_exclude:
"docs/**"

Comment (Lasica) [reply .github/workflows/test.yml:10]: To be clear, according to documentation:
```
When all the path names match patterns in paths-ignore, the workflow will not run. If any path names do not match patterns in paths-ignore, even if some path names match the patterns, the workflow will run.
```

File: .github/workflows/test.yml (hunk 2 above)
Comment (Lasica) [reply .github/workflows/test.yml]: Ok, done

**Review by em-pe — APPROVED** [review]

**Pull request merged**